### ISSUE TITLE
API call /api/core/firmware/status does not trigger check anymore

### ISSUE THREAD

**Issue by tessus** [opnsense/core#6097]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

I am running a [script](https://gist.github.com/tessus/cdbd2367f02211be9224b756b1ecddae) twice a week to check for firmware updates. This script worked perfectly until I upgraded from 22.1.x to 22.7.x.

Turns out that whenver I use the API `/api/core/firmware/status`, it never triggers an update check. I have to login to the UI and click in the dashboard on `Click to check for updates.` or `check for pending updates`.
Only then the script shows the updates.

However, this renders the script useless, if I first have to manually trigger the check in the UI. :-( But even after the check the API call returns the wrong info.

When the UI says `check for pending updates` the response of the API call is:

```
{"api_version": "2", "connection": "ok", "downgrade_packages": [], "download_size": "", "last_check": "Wed Oct 12 00:12:35 EDT 2022", "needs_reboot": "0", "new_packages": [], "os_version": "FreeBSD 13.1-RELEASE-p2", "product_id": "opnsense", "product_target": "opnsense", "product_version": "22.7.5", "reinstall_packages": [], "remove_packages": [], "repository": "ok", "upgrade_major_message": "", "upgrade_major_version": "", "upgrade_needs_reboot": "0", "upgrade_packages": [], "upgrade_sets": [], "product": {"product_abi": "22.7", "product_arch": "amd64", "product_check": {"api_version": "2", "connection": "ok", "downgrade_packages": [], "download_size": "", "last_check": "Wed Oct 12 00:12:35 EDT 2022", "needs_reboot": "0", "new_packages": [], "os_version": "FreeBSD 13.1-RELEASE-p2", "product_id": "opnsense", "product_target": "opnsense", "product_version": "22.7.5", "reinstall_packages": [], "remove_packages": [], "repository": "ok", "upgrade_major_message": "", "upgrade_major_version": "", "upgrade_needs_reboot": "0", "upgrade_packages": [], "upgrade_sets": []}, "product_copyright_owner": "Deciso B.V.", "product_copyright_url": "https://www.deciso.com/", "product_copyright_years": "2014-2022", "product_crypto": "OpenSSL", "product_email": "project@opnsense.org", "product_flavour": "OpenSSL", "product_hash": "5d65a72c6", "product_id": "opnsense", "product_latest": "22.7.6", "product_log": 1, "product_mirror": "https://pkg.opnsense.org/FreeBSD:13:amd64/22.7", "product_name": "OPNsense", "product_nickname": "Powerful Panther", "product_repos": "OPNsense", "product_series": "22.7", "product_time": "Tue Oct 11 15:00:46 EDT 2022", "product_version": "22.7.5", "product_website": "https://opnsense.org/"}, "all_packages": [], "all_sets": [], "status_msg": "There are no updates available on the selected mirror.", "status": "none"}
```

When the UI says `Click to check for updates.` the response of the API call is:

```
{"product": {"product_abi": "22.7", "product_arch": "amd64", "product_check": null, "product_copyright_owner": "Deciso B.V.", "product_copyright_url": "https://www.deciso.com/", "product_copyright_years": "2014-2022", "product_crypto": "OpenSSL", "product_email": "project@opnsense.org", "product_flavour": "OpenSSL", "product_hash": "2eef9015b", "product_id": "opnsense", "product_latest": "22.7.6", "product_license": [], "product_log": 1, "product_mirror": "https://pkg.opnsense.org/FreeBSD:13:amd64/22.7", "product_name": "OPNsense", "product_nickname": "Powerful Panther", "product_repos": "OPNsense", "product_series": "22.7", "product_time": "Tue Oct 18 09:48:00 EDT 2022", "product_version": "22.7.6", "product_website": "https://opnsense.org/"}, "status_msg": "Firmware status requires to check for update first to provide more information.", "status": "none"}
```

**To Reproduce**

1. run script or make API call to `/api/core/firmware/status` (when update available)

**Expected behavior**

Get a rsponse with the updates

**Describe alternatives you considered**

I used the UI and the UI showed the proper updates. But the whole idea of an API call is not to use the UI. ;-)

**Environment**

OPNsense 22.7.6-amd64
FreeBSD 13.1-RELEASE-p2
OpenSSL 1.1.1q 5 Jul 2022

Intel(R) Core(TM) i3-4010U CPU @ 1.70GHz (2 cores, 4 threads)

This issue happens since 22.7.1

**Comment by fichtner** [comment]
Use a POST request if you need a synchronous check…


Cheers,
Franco

**Comment by fichtner** [comment]
The change in question to not start the `check` run during `status` was carried out in 21.1.3 to fix inconsistencies with the behaviour of `check` vs. `upgrade` actions and the subtleties are hard to document in a flat API document generated from code... 

https://github.com/opnsense/changelog/blob/master/community/21.1/21.1.3#L5-L11

That's 21.1, not 22.1 as you suggested. The POST fix was following with 21.1.4:

https://github.com/opnsense/changelog/blob/master/community/21.1/21.1.4#L46


Cheers,
Franco

**Comment by tessus** [comment]
Interesting. Now I am even more confused why the script worked with 22.1.

Unfortunately I deleted all the emails the script sent me before I upgraded to 22.7.
